Makefile

```diff
@@ -18,12 +18,12 @@ SILENT ?= @
 #  usage: "path/to/ignored|another/path"
 UNIT_TEST_IGNORE := "stackrox/rox/sensor/tests"
 
-ifeq ($(TAG),)
-TAG=$(shell git describe --tags --abbrev=10 --dirty --long --exclude '*-nightly-*')
+ifeq ($(BUILD_TAG),)
+BUILD_TAG=$(shell git describe --tags --abbrev=10 --dirty --long --exclude '*-nightly-*')
 endif
```

Comment:
After looking at `scripts/check-debugger.sh`, I remember why there was a difference between `BUILD_TAG` (former `CIRCLE_TAG`) and just `TAG` (that was present in this `Makefile` before edits) in the context of `Makefile`.

* `BUILD_TAG`/`CIRCLE_TAG` meant a tag provided externally, i.e. from CI when we need to build something like `3.74.1-rc.1`.
* `TAG` held the current value of the tag.
  * If `BUILD_TAG`/`CIRCLE_TAG` was provided, `TAG` simply got its value.
  * Otherwise, `TAG` held a result of `git describe`, i.e. some `3.74.x-266-gb6315d348b` thingy.

To me this explains the existence of that `TAG=` trick in `make/env.mk` (intent, but not a reasoning).

Without such separation, we'll not be able to differentiate tagged (i.e. release, RC, nightly) and untagged builds. It seems to me that `scripts/check-debugger.sh` is the only place that does that.
Do you know other places?
How broken is the debugger support that we can leave `check-debugger.sh` in incorrect state?

Reply:
> It seems to me that scripts/check-debugger.sh is the only place that does that.
> Do you know other places?

Here's a deeper example. Our `Makefile` calls into itself, e.g. in case of `main-build-dockerized` target
https://github.com/stackrox/stackrox/blob/d8f6253dd83a7e5d03dbcc8ab0060ff3da1e0799/Makefile#L404-L408

The fact that `BUILD_TAG` **environment variable** is provided highlights a nice distinction that:
* Every environment variable will be visible as makefile variable with the same name. `BUILD_TAG=blah make ...` will do `$(BUILD_TAG)` equal to `blah` during the `Makefile` execution.
* However, it does not work the other way. Makefile variable does not become an environment variable with the same name. I.e. when `${BUILD_TAG}` is not defined in the environment, `main-build-dockerized` propagates no `BUILD_TAG` value into `docker run` context.

This means, `main-build-dockerized` should still work correctly w.r.t. actual `GOTAGS=release` (or not) in the resulting binaries, but the part **why** it works correctly is a bit too subtle.

WDYT, should we keep things the way they are now or should we restore `BUILD_TAG`-`TAG` duality in the `Makefile` for better explicitness?

Looking at it from another angle, the "`BUILD_TAG`-`TAG` duality" was to differentiate when we should and shouldn't apply `GOTAGS=release` during Go binaries build. Maybe that's the place that should be changed to make rules more explicit and straightforward?

Reply:
> WDYT, should we keep things the way they are now or should we restore `BUILD_TAG`-`TAG` duality in the `Makefile` for better explicitness?

I think I will put it back the way it was i.e. `TAG` in the make environment.